libtenzir/builtins/operators/show.cpp

```diff
@@ -0,0 +1,752 @@
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "tenzir/adaptive_table_slice_builder.hpp"
+#include "tenzir/collect.hpp"
+
+#include <tenzir/actors.hpp>
+#include <tenzir/argument_parser.hpp>
+#include <tenzir/atoms.hpp>
+#include <tenzir/catalog.hpp>
+#include <tenzir/error.hpp>
+#include <tenzir/logger.hpp>
+#include <tenzir/node_control.hpp>
+#include <tenzir/partition_synopsis.hpp>
+#include <tenzir/plugin.hpp>
+#include <tenzir/table_slice_builder.hpp>
+#include <tenzir/uuid.hpp>
+
+#include <caf/scoped_actor.hpp>
+#include <caf/timespan.hpp>
+
+#include <string>
+#include <vector>
+
+using namespace tenzir::si_literals;
+
+namespace tenzir::plugins::show {
+
+namespace {
+
+/// Checks whether a thing is an a collection of named things.
+auto contains(const auto& xs, const auto& name) -> bool {
+  auto f = [&](const auto& x) {
+    return x->name() == name;
+  };
+  return std::find_if(xs.begin(), xs.end(), f) != xs.end();
+};
+
+/// A type that represents a connector.
+auto connector_type() -> type {
+  return type{
+    "tenzir.connector",
+    record_type{
+      {"name", string_type{}},
+      {"loader", bool_type{}},
+      {"saver", bool_type{}},
+    },
+  };
+}
+
+/// A type that represents a format.
+auto format_type() -> type {
+  return type{
+    "tenzir.format",
+    record_type{
+      {"name", string_type{}},
+      {"printer", bool_type{}},
+      {"parser", bool_type{}},
+    },
+  };
+}
+
+/// A type that represents an operator.
+auto operator_type() -> type {
+  return type{
+    "tenzir.operator",
+    record_type{
+      {"name", string_type{}},
+      {"source", bool_type{}},
+      {"transformation", bool_type{}},
+      {"sink", bool_type{}},
+    },
+  };
+}
+
+/// A type that represents a partition.
+auto partition_type() -> type {
+  return type{
+    "tenzir.partition",
+    record_type{
+      {"uuid", string_type{}},
+      {"memory_usage", uint64_type{}},
+      {"min_import_time", time_type{}},
+      {"max_import_time", time_type{}},
+      {"version", uint64_type{}},
+      {"schema", string_type{}},
+    },
+  };
+}
+
+/// A type that represents a type attribute.
+auto attribute_type() -> type {
+  return type{
+    "tenzir.attribute",
+    record_type{
+      {"key", string_type{}},
+      {"value", string_type{}},
+    },
+  };
+}
+
+/// A type that describes a type within a schema.
+auto field_type() -> type {
+  return type{
+    "tenzir.field_type",
+    record_type{
+      {"kind", string_type{}},
+      {"category", string_type{}},
+      {"lists", uint64_type()},
+      {"name", string_type{}},
+      {"attributes", list_type{attribute_type()}},
+    },
+  };
+}
+
+/// A type that describes all fields in a schema.
+auto schema_type() -> type {
+  return type{
+    "tenzir.schema",
+    record_type{
+      {"schema", string_type{}},
+      {"fingerprint", string_type{}},
```

Comment:
This should be called `schema_id`, because that's the name we use everywhere else.